x-pack/plugins/file_upload/public/components/index_settings.js

```diff
@@ -0,0 +1,169 @@
+import { getExistingIndices, getExistingIndexPatterns }
+  from '../util/indexing_service';
+
+export const IndexSettings = injectI18n(function IndexSettings({
+  disabled,
+  indexName,
+  setIndexName,
+  setSelectedIndexType,
+  indexTypes,
+  intl,
+  setHasIndexErrors
+}) {
+  const [indexNames, setIndexNames] = useState(null);
+  const [indexNameError, setIndexNameError] = useState('');
+  const [indexPatterns, setIndexPatterns] = useState(null);
+
+  useEffect(() => {
+    setHasIndexErrors(!!indexNameError);
+  });
+
+  if (!indexNames) {
+    getExistingIndices().then(indices => {
+      const indexNames = indices
+        ? indices.map(({ name }) => name)
+        : [];
+      setIndexNames(indexNames);
+    });
+  }
+
+  if (!indexPatterns) {
+    getExistingIndexPatterns().then(
+      indexPatterns => setIndexPatterns(indexPatterns)
+    );
+  }
+
+  const indexDisabled = disabled || !indexTypes || !indexTypes.length;
+
+  return (
+    <Fragment>
+      <EuiSpacer size="m" />
+      <EuiFormRow
+        label={
+          <FormattedMessage
+            id="xpack.file_upload.indexNameLabel"
+            defaultMessage="Index type"
+          />
+        }
+      >
+        <EuiSelect
+          disabled={indexDisabled}
+          options={indexTypes.map(indexType =>({
+            text: indexType,
+            value: indexType,
+          }))}
+          onChange={({ target }) => setSelectedIndexType(target.value)}
+        />
+      </EuiFormRow>
+      <EuiSpacer size="m" />
+      {indexDisabled
+        ? null
+        : (
+          <EuiCallOut
+            title="Index name guidelines"
+            iconType="pin"
+          >
+            <div>
+              <ul>
+                <li>Must be a new index</li>
+                <li>Lowercase only</li>
+                <li>{`Cannot include \\, /, *, ?, ", <, >, |, \` \` \
+                  (space character), , (comma), #`
+                }
+                </li>
+                <li>{`Cannot start with -, _, +`}</li>
+                <li>{`Cannot be . or ..`}</li>
+                <li>{
+                  `Cannot be longer than 255 bytes (note it is bytes, \
+                  so multi-byte characters will count towards the 255 \
+                  limit faster)`
+                }
+                </li>
+              </ul>
+            </div>
+          </EuiCallOut>
+        )}
+      <EuiSpacer size="s" />
+      <EuiFormRow
+        label={
+          <FormattedMessage
+            id="xpack.file_upload.indexNameLabel"
+            defaultMessage="Index name"
+          />
+        }
+        isInvalid={indexNameError !== ''}
+        error={[indexNameError]}
+      >
+        <EuiFieldText
+          disabled={indexDisabled}
+          placeholder={intl.formatMessage({
+            id: 'xpack.file_upload.indexNamePlaceholder',
+            defaultMessage: 'Enter Index Name'
+          })}
+          value={indexName}
```

Comment:
On initial load, suggest index name based on the file-name (providing no conflicts with existing index-patterns of course)